include/spdlog/mdc.h

```diff
@@ -0,0 +1,30 @@
+#include <spdlog/common.h>
+
+namespace spdlog {
+
+class SPDLOG_API mdc {
+public:
+    static void put(const std::string &key, const std::string &value) {
+        get_context()[key] = value;
+    }
+
+    static std::string get(const std::string &key) {
+        auto &context = get_context();
+        auto it = context.find(key);
+        if (it != context.end()) {
+            return it->second;
+        }
+        return "";
+    }
+
+    static void remove(const std::string &key) { get_context().erase(key); }
+
+    static void clear() { get_context().clear(); }
+
+    static std::map<std::string, std::string> &get_context() {
```

Comment:
I think `get_context()` better be private(at least till we find out it is needed by users).

Reply:
get_context() is used in pattern_formatter-inl.h to retrive the mdc map